LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaRequest.cs

```diff
@@ -61,6 +61,9 @@ public virtual void NotifySucceeded(LoRaDevice loRaDevice, DownlinkPktFwdMessage
 
         internal void SetStationEui(StationEui stationEui) => StationEui = stationEui;
 
+        internal static bool RequiresConfirmation(LoRaRequest loraRequest)
+            => loraRequest.Payload is LoRaPayloadData payload && (payload.IsConfirmed || payload.IsMacAnswerRequired);
```

Comment:
why not make this part of the LoRaRequest, overwrite in LoRaPayloadData only and return false for the others? That way you can avoid the casting and we have it controlled by the LoRaPayloadData directly?

Reply:
Good idea, done in 7d7d04a1b48d186903bb1e9878715632d2f2dd0c